sklearn/compose/_column_transformer.py

```diff
@@ -350,8 +350,9 @@ def get_feature_names(self):
         """
         check_is_fitted(self)
         feature_names = []
-        for name, trans, _, _ in self._iter(fitted=True):
-            if trans == 'drop':
+        for name, trans, column, _ in self._iter(fitted=True):
+            if trans == 'drop' or (
+                    hasattr(column, '__len__') and not len(column)):
```

Comment:
```suggestion
                    not column):
```

Reply:
`not column: continue` fails existing tests because `column` can be an index 0..

Reply:
Hmm... Okay I still think this is ugly :)